src/Symfony/Component/Lock/Key.php

```diff
@@ -72,28 +72,38 @@ public function getState($stateKey)
         return $this->state[$stateKey];
     }
 
+    public function resetLifetime()
+    {
+        $this->expiringTime = null;
+    }
+
     /**
      * @param float $ttl The expiration delay of locks in seconds.
      */
     public function reduceLifetime($ttl)
     {
-        $newExpiringDate = \DateTimeImmutable::createFromFormat('U.u', (string) (microtime(true) + $ttl));
+        $newTime = microtime(true) + $ttl;
 
-        if (null === $this->expiringDate || $newExpiringDate < $this->expiringDate) {
-            $this->expiringDate = $newExpiringDate;
+        if (null === $this->expiringTime || $this->expiringTime > $newTime) {
+            $this->expiringTime = $newTime;
         }
     }
 
-    public function resetExpiringDate()
+    /**
+     * Returns the remaining lifetime in seconds.
+     *
+     * @return float|null
+     */
+    public function getRemainingLifetime()
     {
-        $this->expiringDate = null;
+        return null === $this->expiringTime ? null : $this->expiringTime - microtime(true);
```

Comment:
Should it be `null` or `(float) PHP_MAX_INT`?

Reply:
I'd keep null with a comment explaining its meaning in the return annotation

Reply:
👍

Reply:
fixed